tictactoe/game2.rb

```diff
@@ -25,36 +28,31 @@
 
 while turn > 0
   puts
-  puts grid
+  grid_rows.each do |row|
+    puts row
+  end
   puts
 
-  input = $stdin.gets[0]
-  turn = turn-1
+  input = $stdin.gets[0].to_i
 
+  # now that we have input from 1..9,
+  # calculate which row and column 
+  row_number = (input-1) / 3
+  column_number = (input-1) % 3
 
-    row1.split("").each.with_index do |character, index|
-      if character == input
-        grid_row1[index] = turn.odd? ? toe : tictac
-      end
+  player = turn.even? ? tictac : toe
```

Comment:
This makes sense - adding a mother variable for the player based on whether the turn v is even or not... smort!